components/collector/src/source_collectors/jenkins/pipeline_duration.py

```diff
@@ -0,0 +1,73 @@
+"""Jenkins CI-pipeline duration collector."""
+
+from collections.abc import Sequence
+from datetime import timedelta
+from typing import TypedDict, cast
+
+from base_collectors import SourceCollector
+from collector_utilities.date_time import minutes
+from collector_utilities.functions import match_string_or_regular_expression
+from collector_utilities.type import URL, Value
+from model import SourceResponses
+
+from .base import Build, Job
+
+
+class JSON(TypedDict):
+    """Jenkins JSON."""
+
+    builds: Sequence[Build]  # Workflow jobs have builds
+    jobs: Sequence[Job]  # Multibranch pipelines have jobs that have builds
+
+
+class JenkinsPipelineDuration(SourceCollector):
+    """Jenkins CI-pipeline duration collector."""
+
+    async def _api_url(self) -> URL:
+        """Extend to add the job API path and parameters."""
+        url = await super()._api_url()
+        pipeline_job = self._parameter("pipeline")
+        builds = "builds[duration,result,url]"
+        return URL(f"{url}/job/{pipeline_job}/api/json?tree=jobs[name,{builds}],{builds}")
+
+    async def _landing_url(self, responses: SourceResponses) -> URL:
+        """Extend to return the selected build as landing URL if the pipeline has any, or else the pipeline itself."""
+        if build := await self._build(responses):
+            return URL(build["url"])
+        return URL(await super()._landing_url(responses) + f"/job/{self._parameter('pipeline')}")
+
+    async def _parse_value(self, responses: SourceResponses) -> Value:
+        """Parse the value from the responses."""
+        build = await self._build(responses)
+        return str(minutes(timedelta(milliseconds=build["duration"]) if build else timedelta()))
+
+    async def _build(self, responses: SourceResponses) -> Build | None:
+        """Return the selected build, if any."""
```

Comment:
Shouldn't these methods below belong the superclass, or do the other collectors implement it differently?

Reply:
This code is not similar enough to merge into the base class I think.